### PR TITLE
Add support for mbbiDirect and mbboDirect record types

### LINKED ISSUE #33: ENH: Support for mbbi/mbbo record types
In many cases it may not be desirable to generate individual records for each channel on a device, especially for something as simple as digital I/O.

mbbo and mbbi record support should be added to the module for EL1XXX and EL2XXX device support types. These records support up to 16 channels per record, which can majorly reduce clutter.

Support is currently WIP on the `enh_mbbi` branch on my fork.

### PR DESCRIPTION
**Pull request by JJL772**
Initially I was going to add support for mbbi and mbbo, but the direct record types seem to be far more useful for our applications at SLAC.

Most of the changes here are in the record templates, you'll probably want to review commit-by-commit. 

Relies on #32 getting merged first.

Closes #33

### CODE REVIEW

File: ek9000App/src/scripts/generate.py
Comment (klauer) [inline ek9000App/src/scripts/generate.py]: I think `PINI` configured to true for an output record may be in error. Am I misunderstanding? (or is my record behavior faulty? It's been a while since I've done much...)

As I read it, at startup of any given IOC, it will attempt to write the current value (0) for all output records using EK9K. That is to say, all digital outputs would be instructed to go to 0.

File: ek9000App/src/scripts/generate.py
Comment (klauer) [inline ek9000App/src/scripts/generate.py:141]: Not related to this line - for outputs, it'd be nice to have both setpoint and readback records.
Setpoint for what the user intends to write and readback for what the PLC currently reports.

For all of our recent EPICS IOCs (well, at least auto-generated ones) we've followed the AreaDetector convention of having records named as such:
```
pvname       <- the setpoint value
pvname_RBV   <- the corresponding readback value
```

It shouldn't be too difficult to add that record here, I'd think.

If you think it's a reasonable thing to add, let's put it out to another PR though - this one's big enough as is!

File: ek9000App/src/devEK9000.cpp
Comment (klauer) [inline ek9000App/src/devEK9000.cpp]: Intentional switch of debug mode on?

**Review by klauer — APPROVED** [review]
I think this support is a nice improvement 👍 

The device support logic changes look good (admittedly though my C++ is also rusty these days so it took me a bit to read). I didn't spot any real issues with the additional template/substitution files.

File: ek9000App/src/scripts/generate.py
Comment (JJL772) [reply ek9000App/src/scripts/generate.py]: Right, okay. I interpreted the comment as "YES seems like an invalid choice"

I'll take a look at the docs then

File: ek9000App/src/devEK9000.cpp
Comment (JJL772) [reply ek9000App/src/devEK9000.cpp]: Not intentional at all- good catch!

File: ek9000App/src/scripts/generate.py
Comment (JJL772) [reply ek9000App/src/scripts/generate.py]: I think your analysis is correct here. I suspect this hasn't been reported as an issue because most or all of our (admittedly few) ek9k deployments have the watchdog timer enabled with set-to-zero, so outputs get set to 0 regardless when restarting the IOC.

**Comment by klauer** [comment]
Checked the recent commits and looks good to me 👍 
A couple of failing builds on CI though that I didn't dig into, though. Want to check those out before we merge?

**Comment by JJL772** [comment]
I looked at it yesterday, seems to be unrelated to any of my changes. Motor module recently updated all device support submodules, I think they broke builds in one of those device support modules.

For now I just disabled motor, I'll re-enable it on my motor support branch.

**Comment by klauer** [comment]
Ah, yeah, it does seem unrelated to the changes. I'd say this is ready to merge when you're happy with it 👍

**Pull request merged**
